### PR TITLE
swift 3.0 migration

### PR DESCRIPTION
**Pull request by asalom**

### CODE REVIEW

File: SwiftFilePath/PathExtensionDir.swift [hunk 1]
```diff
@@ -21,66 +21,66 @@ extension Path {
...
     }
     
     public class var cacheDir:Path {
-        return Path.userDomainOf(.CachesDirectory)
+        return Path.userDomainOf(.cachesDirectory)
     }
     
-    private class func userDomainOf(pathEnum:NSSearchPathDirectory)->Path{
-        let pathString = NSSearchPathForDirectoriesInDomains(pathEnum, .UserDomainMask, true)[0] 
+    fileprivate class func userDomainOf(_ pathEnum:FileManager.SearchPathDirectory)->Path{
```
Comment (asalom) [inline SwiftFilePath/PathExtensionDir.swift]: made private

File: SwiftFilePath/Path.swift [hunk 2]
```diff
@@ -6,21 +6,21 @@
...
-public class Path {
+open class Path {
     
     // MARK: - Class methods
     
-    public class func isDir(path:NSString) -> Bool {
+    open class func isDir(_ path:String) -> Bool {
         var isDirectory: ObjCBool = false
-        _ = NSFileManager.defaultManager().fileExistsAtPath(path as String, isDirectory:&isDirectory)
+        _ = FileManager.default.fileExists(atPath: path, isDirectory:&isDirectory)
```
Comment (vittoriom) [inline SwiftFilePath/Path.swift:15]: this API is still this ugly even in Swift 3?

Comment (asalom) [inline SwiftFilePath/Path.swift]: I guess I can use this other one: https://developer.apple.com/reference/foundation/filemanager/1415645-fileexists

Comment (asalom) [reply SwiftFilePath/Path.swift:15]: unfortunately yes

**Review by vittoriom — APPROVED** [review]

**Pull request merged**